rfc/plugins.md

```diff
@@ -0,0 +1,338 @@
+There is no hard rule on what technology should be used for the artifact repository, but initial development will use basic HTTP endpoints.
+The download mechanism will be built into the PF4J integration within kork, either using [pf4j/pf4j-update](https://github.com/pf4j/pf4j-update) or a similar system that fits our needs.
+
+A tightly-coupled service and plugin deployment is a poor developer and operator story, but an acceptable shortcoming for an initial release.  There are no current plans for supporting a world where in-process plugins can be upgraded at runtime. There is a lot of complexity and a broad spectrum of potential bugs by hot-reloading plugins, so we aim to avoid this issue entirely by not supporting this use case. If true decoupling of service and plugin is desired, the plugin must be developed and deployed as a remote plugin.
+
+#### Remote
+Remote plugin upgrade process is not much different than iterating a remote service, since that’s what it is.
+
+When a remote plugin is released, its internal functionality will change immediately, just as it would be expected for a remote service. 
+When the remote plugin starts, it must send a registration event to Spinnaker. 
+This registration event would contain, at least:
+
+* Plugin ID
+* Other plugin metadata (author, description, version, etc).
+* What extension points it integrates with
+
+Upon receiving this registration event, Spinnaker would then update the Front50 registry with the new information and downstream services would re-bootstrap the plugin, adding or removing it from extension points as necessary.
+
+### Deployment of Remote Plugins
+Remote plugins can be deployed using whatever means the plugin developer desires.
+In the case of a remote plugin being developed by a third-party outside of the organization, the operator may need to deploy and maintain the plugin themselves.
+Halyard can be used for registering remote plugins with Front50.
+In the absence of Halyard, registering remote plugins with Front50 would be done through the API, just as Halyard would do.
+
+### Deployment of In-Process Plugins
+Downloading plugin resources is necessary for in-process plugins.
+Services need to have plugin resources available locally in order to load them into their process.
+Plugin resources are hosted on a remote server and must be downloaded in order to use them.
+The existing method of downloading plugin resources involves the service itself downloading plugins via a given URI.
+
+#### Kubernetes Environment
+Halyard can be used for updating Front50 with the updated plugin metadata and re-deploying the associated Spinnaker services.
+
+Plugin resources will be placed in the following locations:
+
+- Java services: `/opt/spinnaker/plugins/resources`
+- Deck: `/opt/deck/html/plugins`
+
+Plugins are placed in a different location for Deck so that it can serve plugin resources.
+
+#### VM Based Environment
+Plugins and their configurations are baked into image.
```

Comment:
Doesn't this directly contradict the paragraph on lines 245-248? If they're dowloadable at runtime, by definition they do not need to be baked into any images, no?

Reply:
They do not need to be, no, but they can be. I'll update to be a bit more clear.